subsys/bluetooth/host/classic/hfp_ag.c

```diff
@@ -0,0 +1,2092 @@
+	}
+
+	return k_fifo_get(&ag_tx_free, K_FOREVER);
+}
+
+static void bt_ag_tx_free(struct bt_ag_tx *tx)
+{
+	LOG_DBG("Free tx buffer %p", tx);
+
+	(void)memset(tx, 0, sizeof(*tx));
+
+	k_fifo_put(&ag_tx_free, tx);
+}
+
+static int hfp_ag_next_step(struct bt_hfp_ag *ag, bt_hfp_ag_tx_cb_t cb, void *user_data)
+{
+	struct bt_ag_tx *tx = NULL;
+
+	tx = bt_ag_tx_alloc();
+
+	if (tx == NULL) {
+		LOG_ERR("No Buffers!");
+		return -ENOMEM;
+	}
+
+	tx->ag = ag;
+	tx->buf = NULL;
+	tx->cb = cb;
+	tx->user_data = user_data;
+
+	k_fifo_put(&ag_tx_notify, tx);
+
+	return 0;
+}
+
+static int hfp_ag_send(struct bt_hfp_ag *ag, struct bt_ag_tx *tx)
+{
+	return bt_rfcomm_dlc_send(&ag->rfcomm_dlc, tx->buf);
+}
+
+static int hfp_ag_send_data(struct bt_hfp_ag *ag, bt_hfp_ag_tx_cb_t cb, void *user_data,
+			    const char *format, ...)
+{
+	struct net_buf *buf;
+	struct bt_ag_tx *tx = NULL;
+	va_list vargs;
+	int ret;
+	unsigned int key;
+
+	LOG_DBG("AG %p sending data cb %p user_data %p", ag, cb, user_data);
+
+	buf = bt_rfcomm_create_pdu(&ag_pool);
+	if (!buf) {
+		LOG_ERR("No Buffers!");
+		return -ENOMEM;
+	}
+
+	tx = bt_ag_tx_alloc();
+
+	if (tx == NULL) {
+		LOG_ERR("No Buffers!");
+		net_buf_unref(buf);
+		return -ENOMEM;
+	}
+
+	LOG_DBG("buf %p tx %p", buf, tx);
+
+	tx->ag = ag;
+	tx->buf = buf;
+	tx->cb = cb;
+	tx->user_data = user_data;
+
+	va_start(vargs, format);
+	ret = vsnprintk((char *)buf->data, (net_buf_tailroom(buf) - 1), format, vargs);
+	va_end(vargs);
+
+	if (ret < 0) {
+		LOG_ERR("Unable to format variable arguments");
+		net_buf_unref(buf);
+		bt_ag_tx_free(tx);
+		return ret;
+	}
+
+	net_buf_add(buf, ret);
+
+	key = irq_lock();
+	sys_slist_append(&ag->tx_pending, &tx->node);
+	irq_unlock(key);
+
+	ret = hfp_ag_send(ag, tx);
+
+	if (ret < 0) {
+		LOG_ERR("Rfcomm send error :(%d)", ret);
+		key = irq_lock();
+		sys_slist_find_and_remove(&ag->tx_pending, &tx->node);
+		irq_unlock(key);
+		net_buf_unref(buf);
+		bt_ag_tx_free(tx);
+		return ret;
+	}
+
+	return 0;
+}
+
+static void skip_space(struct net_buf *buf)
+{
+	while ((buf->len > 0) && (buf->data[0] == ' ')) {
+		(void)net_buf_pull(buf, 1);
+	}
+}
+
+static int get_number(struct net_buf *buf, uint32_t *number)
+{
+	int err = -EINVAL;
+
+	*number = 0;
+
+	skip_space(buf);
+	while ((buf->len > 0)) {
+		if ((buf->data[0] >= '0') && (buf->data[0] <= '9')) {
+			*number = *number * 10 + buf->data[0] - '0';
+			(void)net_buf_pull(buf, 1);
+			err = 0;
+		} else {
+			break;
+		}
+	}
+	skip_space(buf);
+
+	return err;
+}
+
+static bool is_char(struct net_buf *buf, uint8_t c)
+{
+	bool found = false;
+
+	skip_space(buf);
+	if ((buf->len > 0)) {
+		if (buf->data[0] == c) {
+			(void)net_buf_pull(buf, 1);
+			found = true;
+		}
+	}
+	skip_space(buf);
+
+	return found;
+}
+
+static int bt_hfp_ag_brsf_handler(struct bt_hfp_ag *ag, struct net_buf *buf, enum at_cme *err)
+{
+	uint32_t number;
+	int error;
+	bool ret;
+
+	*err = CME_ERROR_AG_FAILURE;
+
+	ret = is_char(buf, '=');
+	if (ret != true) {
```

Comment:
A variable of type `bool` is a valid boolean expression in itself, so explicitly comparing it with `true` or `false` is redundant. That said, you can make this more readable by just removing the `ret` variable completely, i.e.:
```c
	if (!is_char(but, '=')) {
		return -EINVAL;
	}
```